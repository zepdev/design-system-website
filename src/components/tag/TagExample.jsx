import React, { useState } from "react"
import Tag from "@zlab-de/zel-react/Tag"

function TagExample() {
  const [text, setText] = useState("I'm a Tag!")

  return (
    <div>
      <Tag onClick={() => setText("You Clicked the Tag!")}>{text}</Tag>
    </div>
  )
}

export default TagExample
