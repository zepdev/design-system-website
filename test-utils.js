import React from "react"
import { render } from "@testing-library/react"
import { ThemeProvider } from "@material-ui/core/styles"
import theme from "./src/gatsby-theme-material-ui-top-layout/theme"

const AllTheProviders = ({ children }) => {
  return <ThemeProvider theme={theme}>{children}</ThemeProvider>
}

const customRender = (ui, options) =>
  render(ui, { wrapper: AllTheProviders, ...options })

// re-export everything
export * from "@testing-library/react"

// override render method
export { customRender as render }
