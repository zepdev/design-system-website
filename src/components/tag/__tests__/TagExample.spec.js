import React from "react"
import { render, fireEvent, screen } from "../../../../test-utils"
import TagExample from "../TagExample"
import "@testing-library/jest-dom/extend-expect"

describe("TagExample", () => {
  it("renders correctly", () => {
    const component = render(<TagExample />)
    expect(component).toMatchSnapshot()
  })
  it("changes text on click", () => {
    render(<TagExample />)
    fireEvent.click(screen.getByRole("button", { name: /I'm a Tag!/i }))
    expect(screen.getAllByRole("button")[0]).toHaveTextContent(
      "You Clicked the Tag!"
    )
  })
})
