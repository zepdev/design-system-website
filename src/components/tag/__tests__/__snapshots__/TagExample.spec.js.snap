// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TagExample renders correctly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div>
        <span
          class="zep-tag"
          role="button"
        >
          I'm a Tag!
          <button
            aria-label="close"
            class="zep-tag__button"
          >
            <svg
              class="zep-tag__icon"
              height="32"
              viewBox="0 0 32 32"
              width="32"
              xmlns="http://www.w3.org/2000/svg"
            >
              <title>
                zepicons-close
              </title>
              <path
                d="M25.333 8.547l-1.88-1.88-7.453 7.453-7.453-7.453-1.88 1.88 7.453 7.453-7.453 7.453 1.88 1.88 7.453-7.453 7.453 7.453 1.88-1.88-7.453-7.453z"
              />
            </svg>
          </button>
        </span>
      </div>
    </div>
  </body>,
  "container": <div>
    <div>
      <span
        class="zep-tag"
        role="button"
      >
        I'm a Tag!
        <button
          aria-label="close"
          class="zep-tag__button"
        >
          <svg
            class="zep-tag__icon"
            height="32"
            viewBox="0 0 32 32"
            width="32"
            xmlns="http://www.w3.org/2000/svg"
          >
            <title>
              zepicons-close
            </title>
            <path
              d="M25.333 8.547l-1.88-1.88-7.453 7.453-7.453-7.453-1.88 1.88 7.453 7.453-7.453 7.453 1.88 1.88 7.453-7.453 7.453 7.453 1.88-1.88-7.453-7.453z"
            />
          </svg>
        </button>
      </span>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
